File: zz/zz.js
#!/usr/bin/env node

const {readFiles, uts} = require("./lib/helpers");

(function usage() {
  const x = readFiles(["VERSION", "GIT_COMMIT_REF", "BUILD_DATE"]);
  console.log(`zz ${x.VERSION}-${x.GIT_COMMIT_REF}`);
  console.log(`build date: ${uts(x.BUILD_DATE)}\n`);
})();

const fetch = require("node-fetch");

(async function() {
  let res, jsn, m;

  console.log("bandwidth -> daily:");
  res = await fetch("https://zfs.rent/api/v0/user/bandwidth-daily");
  jsn = await res.json();
  m = jsn.map(x => ({[x.date]: {domain: x.domain, gb_consumed: parseFloat(x.gb_consumed)}}))
         .reduce((a,b,{}) => Object.assign(a, b));
  console.table(m);

  console.log("\nbandwidth -> monthly:");
  res = await fetch("https://zfs.rent/api/v0/user/bandwidth-monthly");
  jsn = await res.json();
  m = jsn.map(x => ({[x.month]: {domain: x.domain, gb_consumed: parseFloat(x.gb_consumed)}}))
         .reduce((a,b,{}) => Object.assign(a, b));
  console.table(m);

  console.log("\ndisk drives:");
  res = await fetch("https://zfs.rent/api/v0/user/disks");
  jsn = await res.json();
  console.table(jsn);
})()

File: zz/zz.nim
#!/usr/bin/env -S nim r -d:ssl

import strformat
import strutils
import httpclient
const prefix = "https://zfs.rent/api/v0"

const VERSION        = splitLines(staticRead "./VERSION")[0]
const BUILD_DATE     = staticExec "date"
const GIT_COMMIT_REF = staticExec "git rev-parse --short HEAD"

echo &"zz {VERSION}-{GIT_COMMIT_REF}"
echo &"build date: {BUILD_DATE}\n"

try:
  var client = newHttpClient()
  echo "Requesting daily bandwidth..."
  echo client.getContent(prefix & "/user/bandwidth-daily")
except:
  let e = getCurrentException()
  echo fmt"{e.name} :: {e.msg}"
  quit(QuitFailure)
